src/30_Advanced/SeatMap.html

```diff
@@ -0,0 +1,272 @@
+        stroke-width: 3;
+        stroke: rgb(0,0,0);
+        fill:white;
+      }
+      .unavailable {
+      fill: var(--abe-light-pink);
+      pointer-events: none;
+      }
+      .special-seat {
+      fill: white;
+      }
+      .special-seat-image {
+        stroke-width: 3;
+        stroke: rgb(0,0,0);
+        fill:white;
+      }
+      .selected.special-seat-image {
+      fill: var(--abe-pink);
+      background-color: var(--abe-pink);
+      }
+      .container {
+      margin: 0;
+      display: flex;
+      flex-direction: column;
+      background: white;
+      border-radius: 5px;
+      }
+      .seat.selected {
+      fill: var(--abe-pink);
+      }
+      .disabled-seat {
+      pointer-events: none;
+      cursor: default;
+      }
+      .content {
+      background: white;
+      flex-grow: 1;
+      display: flex;
+      flex-direction: column;
+      border-radius: 5px;
+      }
+      .disabled {
+      pointer-events: none;
+      opacity: 0.5;
+      cursor: default;
+      }
+      .hidden {
+      display: none;
+      }
+      .clickable {
+      cursor: pointer;
+      }
+      .zoom-button {
+
+      bottom: 60px;
+      right: 0px;
+      z-index: 1;
+      border-color: Transparent;
+      background-color: Transparent;
+      margin: 8px;
+      width: 40px;
+      height: 40px;
+      padding: 0;
+      border-width: 0;
+      box-shadow: 0 2px 4px 0 rgba(0, 0, 0, 0.25);
+      background-color: white;
+      border-radius: 3px;
+      }
+      .opaque {
+      opacity: 0.5;
+      filter: alpha(opacity=50);
+      }
+      #choose-seats-button {
+      padding-bottom: 15px;
+      padding-right: 10px;
+      }
+      .seat-map-svg {
+      padding: 0 30px;
+      box-sizing: border-box;
+      }
+      .hide {
+      display: none;
+      }
+      .list-container {
+        position: relative;
+        height: 70vh;
+      }
+      .ampstart-card {
+        min-width: 100%;
+      }
+    </style>
+    <style amp-boilerplate>body{-webkit-animation:-amp-start 8s steps(1,end) 0s 1 normal both;-moz-animation:-amp-start 8s steps(1,end) 0s 1 normal both;-ms-animation:-amp-start 8s steps(1,end) 0s 1 normal both;animation:-amp-start 8s steps(1,end) 0s 1 normal both}@-webkit-keyframes -amp-start{from{visibility:hidden}to{visibility:visible}}@-moz-keyframes -amp-start{from{visibility:hidden}to{visibility:visible}}@-ms-keyframes -amp-start{from{visibility:hidden}to{visibility:visible}}@-o-keyframes -amp-start{from{visibility:hidden}to{visibility:visible}}@keyframes -amp-start{from{visibility:hidden}to{visibility:visible}}</style>
+    <noscript>
+      <style amp-boilerplate>body{-webkit-animation:none;-moz-animation:none;-ms-animation:none;animation:none}</style>
+    </noscript>
+
+  </head>
+  <body>
+    <!-- ## How to implement a seat map -->
+    <!-- Here we are using an `amp-state` called `user`
+    to keep a list of selected seat ids in the `selectedSeats` array -->
+    <amp-state id="user">
```

Comment:
Suggesting: you use JSON objects with a single value which seems too complicated. I see two ways to simplify this, either remove the indirection:

```
<amp-state id="selectedSeats">
      <script type="application/json">[]</script>
</amp-state>
```

or put everything into a single state var:

```
    <amp-state id="seatmap">
       <script type="application/json">
         {
           "scale": 1,
           "selectedSeats": 1,
           "zoom": 0
         }
       </script>
     </amp-state>
```